src/Protocols.Abstractions/PipeFactoryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace System.IO.Pipelines
+{
+    public static class PipeFactoryExtensions
+    {
+        public static (IPipeConnection Transport, IPipeConnection Application) CreateConnectionPair(this PipeFactory pipeFactory)
+        {
+            return pipeFactory.CreateConnectionPair(new PipeOptions(), new PipeOptions());
+        }
+
+        public static (IPipeConnection Transport, IPipeConnection Application) CreateConnectionPair(this PipeFactory pipeFactory, PipeOptions inputOptions, PipeOptions outputOptions)
+        {
+            var input = pipeFactory.Create(inputOptions);
+            var output = pipeFactory.Create(outputOptions);
+
+            var transportToApplication = new PipeConnection(output.Reader, input.Writer);
+            var applicationToTransport = new PipeConnection(input.Reader, output.Writer);
```

Comment:
No, it's too verbose. Also the only thing that usually sees both sides are when you're implementing the transport. Notice the `DefaultConnectionContext` still has the `Transport` exposed.